src/main/kotlin/biokotlin/cli/ValidateAndCorrectGVCFs.kt

```diff
@@ -0,0 +1,107 @@
+package biokotlin.cli
+
+import biokotlin.genome.fastaToNucSeq
+import biokotlin.util.bufferedReader
+import biokotlin.util.bufferedWriter
+import biokotlin.util.parseVCFFile
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import kotlinx.coroutines.runBlocking
+import java.io.File
+
+class ValidateAndCorrectGVCFs : CliktCommand(help = "Validate and correct GVCF files") {
+
+    val inputDir by option(help = "Full path to input GVCF file directory")
+        .required()
+
+    val outputDir by option(help = "Full path to output GVCF file directory")
+        .required()
+
+    val referenceFile by option(help = "Full path to reference fasta file")
+        .required()
+
+    override fun run() {
+
+        // Checks to ensure that the input and output directories and reference file exist
+        require(File(inputDir).isDirectory) { "Input GVCF directory does not exist: $inputDir" }
+
+        require(File(outputDir).isDirectory) { "Output GVCF directory does not exist: $outputDir" }
+
+        require(File(inputDir).absolutePath != File(outputDir).absolutePath) { "Input and output GVCF directories are the same: $inputDir" }
+
+        require(File(referenceFile).isFile) { "Reference FASTA file does not exist: $referenceFile" }
+
+        // Reference from FASTA file considered the correct reference sequences
+        // Map of <contig, NucSeq>
+        val refSeqGenome = fastaToNucSeq(referenceFile)
+
+        // Get list of input GVCF files from the input directory
+        val inputFiles = File(inputDir)
+            .walk()
+            .filter {
+                it.isFile && (it.name.endsWith(".g.vcf") || it.name.endsWith(".g.vcf.gz") ||
+                        it.name.endsWith(".gvcf") || it.name.endsWith(".gvcf.gz"))
+            }
+            .map { it.absolutePath }
+            .toList()
+
+        // Iterate through each GVCF file from the input directory
+        inputFiles.forEach { inputFile ->
+
+            val outputFile = "$outputDir/${File(inputFile).name}"
+            val logFile = "$outputDir/${File(inputFile).name}.log"
+
+            println()
+            println("Input GVCF file: $inputFile")
+            println("Output GVCF file: $outputFile")
+            println("Log file: $logFile")
```

Comment:
Is this just for debugging or should this be in the logging?

Reply:
I changed this to a logger